ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,226 @@
+fi
+
+while IFS= read -r line; do
+    if [[ "$line" != *"ConnectX"* ]]; then continue; fi
+    mlnx_type="${line##*-}"
+    mlnx_type="${mlnx_type%*]}"
+    if [ "$mlnx_type" -ge 6 ]; then
+        ((hdr_count=hdr_count+1))
+    fi
+done < <(lspci | grep "ConnectX")
+if [ $hdr_count -gt 0 ]; then
+    ((ib_count=hdr_count-1)) || true
+fi
+
+# Can not have Omni-Path and Mellanox HDR on the same system.
+if [ $hdr_count -gt 0 ] && [ $opa_count -gt 0 ]; then
+    ib_message="$({
+        echo "Invalid hardware configuration.  Found:"
+        echo "$hdr_count Mellanox HDR ConnectX adapters."
+        echo "and"
+        echo "$opa_count Omni-Path adapters"
+        echo "The Onmi-Path adapters will not be used."
+        })"
+    mail_message+="${nl}${ib_message}${nl}"
+    echo "$ib_message"
+fi
+set -x
+
+function do_wait_for_ib {
+    ib_timeout=300 # 5 minutes
+    retry_wait=10 # seconds
+    timeout=$((SECONDS + ib_timeout))
+    while [ "$SECONDS" -lt "$timeout" ]; do
+      ip link set up "$1" || true
+      sleep 2
+      if ip addr show "$iface" | grep "inet "; then
+        return 0
+      fi
+      sleep ${retry_wait}
+    done
+}
+
+# First check for infinband devices
+for i in $(seq 0 $ib_count); do
     iface="ib$i"
-    if [ -e /sys/class/net/"$iface" ]; then
-        if ! ifconfig "$iface" | grep "inet "; then
-          {
-            echo "Found interface $iface down after reboot on $HOSTNAME"
-            systemctl status || true
-            systemctl --failed || true
-            journalctl -n 500 || true
-            ifconfig "$iface" || true
-            cat /sys/class/net/"$iface"/mode || true
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+    if do_wait_for_ib "$iface"; then
+        set +x
+        if ! ip addr show "$iface" | grep "inet "; then
+            ib_message="$({
+                echo "Found interface $iface down after reboot on $HOSTNAME"
+                ip addr show "$iface" || true
+                cat /sys/class/net/"$iface"/mode || true
+                ip link set up "$iface" || true
+                cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
+                } 2>&1)"
+            mail_message+="${nl}${ib_message}${nl}"
+            echo "$ib_message"
+        else
+            echo "OK: Interface $iface is up."
         fi
-        if ! ifconfig "$iface" | grep "inet "; then
+        if ! ip addr show "$iface" | grep "inet "; then
+            echo "Fail: $ib_message"
+            result=1
             echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
-            exit 1
+            "Please file a SRE ticket."
```

Comment:
Do we need a second `if ! ip addr show "$iface" | grep "inet "; then`?  Shouldn't the code just go in the first conditional where `$ib_message` is defined?:
```suggestion
            echo "Fail: $ib_message"
            result=1
            echo "Failed to bring up interface $iface on $HOSTNAME. " \
            "Please file a SRE ticket."
        else
            echo "OK: Interface $iface is up."
```